### PR TITLE
test infra: Remove timeSource() from the ClusterManager api

### PR DESCRIPTION
**Pull request by jmarantz**
*Description*: Follow-up to #4180  -- removes ClusterManager::timeSource() as an API. Another step toward resolving #4160 

*Risk Level*: low
*Testing*: //test/...
*Docs Changes*: n/a
*Release Notes*: n/a

### CODE REVIEW

File: source/common/grpc/async_client_impl.h [hunk 1]
```diff
@@ -25,13 +26,14 @@ class AsyncClientImpl final : public AsyncClient {
   AsyncStream* start(const Protobuf::MethodDescriptor& service_method,
                      AsyncStreamCallbacks& callbacks) override;
 
-  TimeSource& timeSource() { return cm_.timeSource(); }
+  TimeSource& timeSource() { return time_source_; }
```
Comment (htuch) [inline source/common/grpc/async_client_impl.h]: Still a bit puzzled why we need to have a getter for `TimeSource` here..

File: source/common/router/router.h [hunk 2]
```diff
@@ -96,26 +96,29 @@ class FilterConfig {
...
+                     config.start_child_span(), config.suppress_envoy_headers(),
+                     context.timeSource()) {
     for (const auto& upstream_log : config.upstream_log()) {
       upstream_logs_.push_back(AccessLog::AccessLogFactory::fromProto(upstream_log, context));
     }
   }
 
   ShadowWriter& shadowWriter() { return *shadow_writer_; }
-  TimeSource& timeSource() { return cm_.timeSource(); }
+  TimeSource& timeSource() { return time_source_; }
```
Comment (htuch) [inline source/common/router/router.h]: ... and I guess here as well.. neither of these are new to this PR, so I think it makes sense to merge and continue the discussion elsewhere.

File: test/common/upstream/load_stats_reporter_test.cc [hunk 3]
```diff
@@ -74,7 +76,9 @@ class LoadStatsReporterTest : public testing::Test {
   Event::TimerCb response_timer_cb_;
   Grpc::MockAsyncStream async_stream_;
   Grpc::MockAsyncClient* async_client_;
-  MockMonotonicTimeSource time_source_;
+  ProdSystemTimeSource system_time_source_;
```
Comment (htuch) [inline test/common/upstream/load_stats_reporter_test.cc]: Why is this prod? Won't your recent fix_format check complain?

Comment (jmarantz) [reply test/common/upstream/load_stats_reporter_test.cc]: It whitelists test/*

Comment (jmarantz) [reply test/common/upstream/load_stats_reporter_test.cc]: It whitelists test/*

Comment (htuch) [reply test/common/upstream/load_stats_reporter_test.cc]: OK, but why not use a mock alternative all things being equal when adding new entries here?

Comment (jmarantz) [reply test/common/upstream/load_stats_reporter_test.cc]: I think that would change behavior if it were called. Current status is that real time is used everywhere in tests where there is not already an explicit mock. These can be cleaned up one time is injectable.

Note in particular that dispatcher uses real time for timers unconditionally, though I have some pushed WIP and unpushed changes I am working that add simulated timers.

File: source/common/grpc/async_client_impl.h (hunk 1 above)
Comment (jmarantz) [reply source/common/grpc/async_client_impl.h]: source/common/grpc/async_client_impl.cc:211 needs this, but I see that class is already a friend, and in this code we already have the pattern of friends digging into private member vars without going thru a private method. So I can remove it.

File: source/common/router/router.h (hunk 2 above)
Comment (jmarantz) [reply source/common/router/router.h]: This is needed in source/common/router/router.cc:782, and there is no current 'friend' declaration, so I prefer to leave it as is; it's not clear why this exposes more than the shadowWriter accessor above.

File: test/common/upstream/load_stats_reporter_test.cc (hunk 3 above)
Comment (jmarantz) [reply test/common/upstream/load_stats_reporter_test.cc]: In any event it doesn't affect this test, and in an upcoming #4257 I'll effectively be making this semantic change anyway, so I went ahead and made the change you suggested.

**Comment by dnoe** [comment]
I'll assign final reviewers to this whenever you're confident the CI failure is addressed.

**Pull request merged**